tests/components/lawn_mower/test_init.py

```diff
@@ -0,0 +1,171 @@
+    MockPlatform,
+    mock_config_flow,
+    mock_integration,
+    mock_platform,
+)
+
+TEST_DOMAIN = "test"
+
+
+class MockFlow(ConfigFlow):
+    """Test flow."""
+
+
+class MockLawnMowerEntity(LawnMowerEntity):
+    """Mock lawn mower device to use in tests."""
+
+    def __init__(
+        self,
+        unique_id: str = "lawn_mower",
+        name: str = "Lawn Mower",
+        features: LawnMowerEntityFeature = LawnMowerEntityFeature(0),
+    ) -> None:
+        """Initialize the lawn mower."""
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_supported_features = features
+
+    def start_mowing(self) -> None:
+        """Start mowing."""
+        self._attr_activity = LawnMowerActivity.MOWING
+
+
+@pytest.fixture(autouse=True)
+def config_flow_fixture(hass: HomeAssistant) -> Generator[None, None, None]:
+    """Mock config flow."""
+    mock_platform(hass, f"{TEST_DOMAIN}.config_flow")
+
+    with mock_config_flow(TEST_DOMAIN, MockFlow):
+        yield
+
+
+async def test_lawn_mower_setup(hass: HomeAssistant) -> None:
+    """Test setup and tear down of lawn mower platform and entity."""
+
+    async def async_setup_entry_init(
+        hass: HomeAssistant, config_entry: ConfigEntry
+    ) -> bool:
+        """Set up test config entry."""
+        await hass.config_entries.async_forward_entry_setup(
+            config_entry, Platform.LAWN_MOWER
+        )
+        return True
+
+    async def async_unload_entry_init(
+        hass: HomeAssistant, config_entry: ConfigEntry
+    ) -> bool:
+        """Unload up test config entry."""
+        await hass.config_entries.async_unload_platforms(
+            config_entry, [Platform.LAWN_MOWER]
+        )
+        return True
+
+    mock_platform(hass, f"{TEST_DOMAIN}.config_flow")
+    mock_integration(
+        hass,
+        MockModule(
+            TEST_DOMAIN,
+            async_setup_entry=async_setup_entry_init,
+            async_unload_entry=async_unload_entry_init,
+        ),
+    )
+
+    entity1 = LawnMowerEntity()
+    entity1.entity_id = "lawn_mower.test1"
+
+    async def async_setup_entry_platform(
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        async_add_entities: AddEntitiesCallback,
+    ) -> None:
+        """Set up test platform via config entry."""
+        async_add_entities([entity1])
+
+    mock_platform(
+        hass,
+        f"{TEST_DOMAIN}.{Platform.LAWN_MOWER}",
+        MockPlatform(async_setup_entry=async_setup_entry_platform),
+    )
+
+    config_entry = MockConfigEntry(domain=TEST_DOMAIN)
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(entity1.entity_id)
+
+    assert await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.config_entries.async_get_entry(config_entry.entry_id) is None
```

Comment:
```suggestion
    assert config_entry.state == ConfigEntryState.NOT_LOADED
```

Reply:
Let's also assert that the platform has been unloaded like so:
https://github.com/home-assistant/core/blob/8b4937f627126f5b6a16776e6def1a5070124641/tests/components/matter/test_init.py#L78-L81

Reply:
Please add an assertion of the entity state after unloading the config entry as in the example above. That way we make sure that we have hooked up the lawn mower integration to the unloading machinery of the platform helper.